hydra_redis/redisgraph_demo1.py

```diff
@@ -0,0 +1,348 @@
+from re import compile as regex
+from httplib2 import Http
+from hydrus.hydraspec import doc_maker
+
+
+redis_con = redis.Redis(host='localhost', port=6379)
+redis_graph = Graph("apidoc", redis_con)
+
+
+class _MemCache(dict):
+
+    def __nonzero__(self):
+        # even if empty, a _MemCache is True
+        return True
+
+    def set(self, key, value):
+        self[key] = value
+
+    def delete(self, key):
+        if key in self:
+            del self[key]
+
+
+DEFAULT_HTTP_CLIENT = Http(_MemCache())
+http = DEFAULT_HTTP_CLIENT
+APIDOC_RE = regex(
+    r'^<([^>]*)>; rel="http://www.w3.org/ns/hydra/core#apiDocumentation"$')
+
+
+def final_file(url):
+    response = ur.urlopen(url)
+    new_file = json.loads(response.read().decode('utf-8'))
+    return new_file
+
+
+def get_method(new_obj):
+    obj_method = []
+    for obj in new_obj:
+        obj_method.append(obj["method"].lower())
+    return obj_method
+
+
+def get_title(new_obj):
+    new_obj = new_obj[0]
+#    print (new_obj)
+    obj_title = new_obj["description"]
+    print(obj_title)
+    return str(obj_title[4:])
+
+
+def get_endpoint_id(string):
+    return (string.replace("vocab:EntryPoint", ""))
+
+
+def get_member_alias(string1, string2):
+    match_obj = re.match(r'/(.*)/(.*)/(.*)?', string1, re.M | re.I)
+    base_url = "/{0}/{1}/".format(match_obj.group(1), match_obj.group(2))
+    entrypoint_member = string2 + match_obj.group(3)
+    print(base_url, entrypoint_member)
+    return entrypoint_member, match_obj.group(3)
+
+
+def remove_vocab(string):
+    return string.replace("vocab:", "")
+
+
+def get_property(new_obj):
+    obj_prop = []
+    for obj in new_obj:
+        for obj1 in apidoc["supportedClass"]:
+            if obj1["@id"] == obj["property"]:
+                print(obj_prop, obj["property"])
+                obj_prop.append(obj["property"])
+    if obj_prop:
+        return obj_prop
+    else:
+        return []
+
+
+def connect_nodes(source_node, predicate, dest_node):
+    edge = Edge(source_node, predicate, dest_node)
+    redis_graph.add_edge(edge)
+    print("add edge commit", predicate)
+    redis_graph.commit()
+    print("edge", edge)
+
+
+def objects_property(objects_node, new_list, url_node):
+    print("for the properties which have class and should have node")
+
+    for obj in new_list:
+        objects_properties = {}
+        for obj1 in apidoc["supportedClass"]:
+            if obj1["@id"] == obj:
+                properties = []
+                if obj1["supportedOperation"]:
+                    objects_properties["methods"] = str(
+                        get_method(obj1["supportedOperation"]))
+                if obj1["supportedProperty"]:
+                    for obj2 in obj1["supportedProperty"]:
+                        properties.append(obj2["title"])
+                    objects_properties["property"] = str(properties)
+                    obj_properties_classlist = get_property(
+                        obj1["supportedProperty"])
```

Comment:
You need to get all these properties from the api_doc you created in line 339.
Let me show you an example.
**You can simply use the APIDoc object to get all the required properties. No need to manually parse the object. See screenshots below**
![selection_021](https://user-images.githubusercontent.com/9114274/40872293-68f4887e-6669-11e8-8b8e-96336f4a232e.png)

![selection_022](https://user-images.githubusercontent.com/9114274/40872294-6947d768-6669-11e8-835b-55ba79ff233f.png)

![selection_023](https://user-images.githubusercontent.com/9114274/40872295-69931ae8-6669-11e8-9789-74bfa2017583.png)

Please analyze the following files and update your code accordingly.
https://github.com/HTTP-APIs/hydrus/blob/master/hydrus/hydraspec/doc_writer.py
https://github.com/HTTP-APIs/hydrus/blob/master/hydrus/hydraspec/doc_maker.py